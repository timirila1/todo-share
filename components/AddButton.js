import React from 'react';
import { View, StyleSheet, TouchableHighlight } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

export const AddButton = props => {
    const { onPress } = props;

    return (
        <TouchableHighlight
            onPress={onPress}
            style={styles.wrapper}
            underlayColor="black"
        >
            <View style={styles.button}>
                <Ionicons
                    name="ios-add"
                    size={40}
                    color="white"
                />
            </View>
        </TouchableHighlight>
    );
};

const styles = StyleSheet.create({
    wrapper: {
        position: 'absolute',
        right: 40,
        bottom: 40,
        borderRadius: 20,
    },
    button: {
        backgroundColor: '#fa6c35',
        paddingTop: 10,
        paddingBottom: 10,
        paddingRight: 20,
        paddingLeft: 20,
        borderRadius: 20,
    }
});
