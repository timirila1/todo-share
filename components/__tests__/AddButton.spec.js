import 'react-native';
import React from 'react';
import { AddButton } from '../AddButton';
import renderer from 'react-test-renderer';

it('Render AddButton', async () => {
    const tree = renderer.create(
        <AddButton />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
