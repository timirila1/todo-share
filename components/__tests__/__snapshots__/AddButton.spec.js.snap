// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render AddButton 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 20,
      "bottom": 40,
      "position": "absolute",
      "right": 40,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fa6c35",
        "borderRadius": 20,
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
      }
    }
  >
    <Text />
  </View>
</View>
`;
